go/adbc/driver/snowflake/connection.go

```diff
@@ -239,13 +261,13 @@ func (c *connectionImpl) GetCurrentDbSchema() (string, error) {
 
 // SetCurrentCatalog implements driverbase.CurrentNamespacer.
 func (c *connectionImpl) SetCurrentCatalog(value string) error {
-	_, err := c.cn.ExecContext(context.Background(), "USE DATABASE ?", []driver.NamedValue{{Value: value}})
+	_, err := c.cn.ExecContext(context.Background(), fmt.Sprintf("USE DATABASE %s", value), nil)
```

Comment:
Bobby Tables says never to forget about quoting SQL identifiers properly. Snowflake identifiers can contain spaces and double-quotes and semicolons, and are quoted by using delimiting with double-quotes and doubling any contained double-quotes. That is, `A` -> `"A"`, `A B` -> `"A B", `A B / C" ; D` -> `"A B / C"" ; D"`. (This is pretty consistent with most SQL dialects.

Reply:
Bah, I've succeeded in confusing the "code" feature of Markdown. That last substitution should be
A B / C" ; D -> "A B / C"" ; D"

Reply:
Could we use snowflake's `IDENTIFIER(...)` function here?

e.g. `fmt.Sprintf("USE DATABASE IDENTIFIER('%s')", value)`

That should eliminate possible injection scenarios, right?

Reply:
This doesn't work, unfortunately. (I tried it specifically, and it's also not usually how SQL grammars work: you can't usually supply an expression here, only an identifier.)

Reply:
(I also feel obliged to point out that any single quotes in the identifier would need to be escaped in that variant.)

Reply:
went with Matt's suggestion

Reply:
I thought we had a proper quote function? Can we use that?

Reply:
we have quoteTblName(...)

Reply:
quoteTblName is what we should be using

Reply:
done